thunderbirdparser/src/org/sleuthkit/autopsy/thunderbirdparser/EmailMessageThreader.java

```diff
@@ -0,0 +1,674 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2019 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.thunderbirdparser;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Given a list of email messages arranges the message into threads using the
+ * message reference lists.
+ *
+ * This threader is based heavely off of the algorithum found at
+ * <a href="https://www.jwz.org/doc/threading.html">
+ * "message threading." by Jamie Zawinski</a>
+ *
+ */
+final class EmailMessageThreader {
+
+    private int bogus_id_count = 0;
+
+    public void threadMessages(List<EmailMessage> emailMessages, String threadIDPrefix) {
```

Comment:
Seems like this should all be in a constructor since the class isn't useful without this data being passed in.

Reply:
Or, make it static since it doesn't really save any state.